packages/malloy/src/lang/parse-malloy.ts

```diff
@@ -783,10 +785,80 @@ export abstract class MalloyTranslation {
     const attempt = this.translateStep.step(this, extendingModel);
     if (attempt.final) {
       this.finalAnswer = attempt;
+      this.buildImports();
     }
     return attempt;
   }
 
+  importAt(position: DocumentPosition): ImportLocation | undefined {
+    // Here we assume that imports DO NOT overlap. And then we do a binary
+    // search to find the one we're looking for.
+    const index = this.findIndexBefore(position);
+    if (index === this.imports.length) {
+      return undefined;
+    }
+    const imp = this.imports[index];
+    if (
+      imp.location.range.start.line <= position.line &&
+      imp.location.range.end.line >= position.line &&
+      (position.line !== imp.location.range.start.line ||
+        position.character >= imp.location.range.start.character) &&
+      (position.line !== imp.location.range.end.line ||
+        position.character <= imp.location.range.end.character)
+    ) {
+      return imp;
+    }
+    return undefined;
+  }
+
+  private buildImports(): void {
+    for (const [key, value] of Object.entries(this.root.importZone.location)) {
+      this.imports.push({importURL: key, location: value});
+    }
+    // Sort list of imports by end position so that we can do a binary
+    // search to find an import at a given position.
+    this.imports.sort((i1, i2) => {
```

Comment:
The binary search stuff doesn't belong in the `parse-malloy.ts` file. Maybe you can make https://github.com/malloydata/malloy/blob/73b05828440fb2ffc1075312283c1e97ef9f9d50/packages/malloy/src/lang/reference-list.ts#L4 generic to support this use case.